### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by kgaillot**
The main goal is to get the cppcheck and stonith_admin changes in before 2.0.4-rc1. The rest were just quick improvements.

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , when you get a chance can you review this? I found a couple of things needing fixing before rc1 so I did "just one more" grab bag of minor fixes.

File: lib/common/io.c [hunk 1]
```diff
@@ -33,35 +33,56 @@
...
 
 /*!
+ * \internal
  * \brief Create a directory, including any parent directories needed
  *
  * \param[in] path_c Pathname of the directory to create
  * \param[in] mode Permissions to be used (with current umask) when creating
  *
- * \note This logs errors but does not return them to the caller.
+ * \return pcmk_ok on success, -errno otherwise
```
Comment (clumens) [inline lib/common/io.c:42]: Would it be worth converting this function to returning a standard pacemaker return code now too?

File: lib/pacemaker/pcmk_fence.c [hunk 2]
```diff
@@ -190,32 +190,34 @@ pcmk__fence_history(pcmk__output_t *out, stonith_t *st, char *target,
                            safe_str_eq(target, "*") ? NULL : target,
                            &history, timeout/1000);
 
-    out->begin_list(out, "event", "events", "Fencing history");
+    if (!cleanup) { // Cleanup doesn't return a history list
```
Comment (clumens) [inline lib/pacemaker/pcmk_fence.c:193]: I'd prefer to see this written more like:
```
if (cleanup) {
    stonith_history_free(history);
    return pcmk_legacy2rc(rc);
}
```

Or whatever.  And then the rest is indented normally.

File: lib/common/io.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/io.c:42]: Whoops, that had been sitting around as a patch for a long time. Will do.

File: GNUmakefile [hunk 3]
```diff
@@ -393,6 +395,7 @@ CPPCHECK_ARGS ?=
 cppcheck:
 	cppcheck $(CPPCHECK_ARGS) -I include --max-configs=30	\
 		--library=posix --library=gnu --library=gtk	\
+		$(GLIB_CFLAGS) -D__GNUC__			\
 		--inline-suppr -q replace lib daemons tools
```
Comment (clumens) [inline GNUmakefile:400]: Oh cool, I'd been meaning to ask you what your cppcheck command line looks like but I guess here it is.

File: lib/pacemaker/pcmk_fence.c (hunk 2 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_fence.c:193]: makes sense

**Comment by kgaillot** [comment]
Updated

**Comment by clumens** [comment]
Looks good.

**Pull request merged**